File: app/src/main/java/id/wisata/RecyclerViewAdapter.java
package id.wisata;

import android.content.Context;
import android.os.Build;
import android.support.annotation.NonNull;
import android.support.annotation.RequiresApi;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import java.util.List;

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewAdapter.MyViewHolder> {


    private Context ctx;
    private List<Place> data;

    public RecyclerViewAdapter(Context ctx, List<Place> data){
        this.ctx = ctx;
        this.data = data;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {

        View view ;
        LayoutInflater inflater = LayoutInflater.from(ctx);
        view = inflater.inflate(R.layout.place_item_card_view, viewGroup, false);

        return new MyViewHolder(view);
    }


    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, int i) {

        myViewHolder.placeName.setText(data.get(i).getName());
        myViewHolder.placePicture.setImageResource(data.get(i).getImageUrl());
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public static  class MyViewHolder extends RecyclerView.ViewHolder {

        TextView placeName;
        ImageView placePicture;

        public MyViewHolder(View itemView) {
            super(itemView);

            placeName = (TextView) itemView.findViewById(R.id.placeName);
            placePicture = (ImageView) itemView.findViewById(R.id.placePicture);
        }
    }
}

File: app/src/main/java/id/wisata/view/adapter/RecyclerViewAdapter.kt
package id.wisata.view.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import id.wisata.R
import id.wisata.data.model.Place
import kotlinx.android.synthetic.main.place_item_card_view.view.*

class RecyclerViewAdapter(private val ctx: Context,
                          private  val items: List<Place>,
                          private val clickListener: OnItemClickListener) :
    RecyclerView.Adapter<RecyclerViewAdapter.ViewHolder>() {


    interface OnItemClickListener{
        fun onItemClick(place: Place, itemView : View)
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {

        val view: View
        val inflater = LayoutInflater.from(ctx)
        view = inflater.inflate(R.layout.place_item_card_view, viewGroup, false)

        return ViewHolder(view)
    }


    override fun onBindViewHolder(holder: ViewHolder, i: Int) {

        (holder).bind(items[i], clickListener)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        fun bind(place: Place, listener: OnItemClickListener) = with(itemView) {
            itemView.placeName.text = place.name
            placePicture.setImageDrawable(resources.getDrawable(place.imageUrl))



            setOnClickListener {
                listener.onItemClick(place, it)
            }
        }
    }
}
